src/Microsoft.IdentityModel.JsonWebTokens/JsonWebToken.cs

```diff
@@ -613,6 +613,15 @@ public virtual IEnumerable<Claim> Claims
             }
         }
 
+        /// <summary>
+        /// Gets an <see cref="IReadOnlyList{Claim}"/> where each claim in the JWT header { name, value } is returned as a <see cref="Claim"/>.
+        /// </summary>
+        /// <remarks>
+        /// A <see cref="Claim"/> requires each value to be represented as a string. If the value was not a string, then <see cref="Claim.Type"/> contains the json type.
+        /// <see cref="JsonClaimValueTypes"/> and <see cref="ClaimValueTypes"/> to determine the json type.
+        /// </remarks>
+        public IReadOnlyList<Claim> HeaderClaims => Header.Claims(Issuer ?? ClaimsIdentity.DefaultIssuer);
```

Comment:
Alternatives: Only expose header names, and then call GetHeaderValue<Claim>(key) with each key from the header names, or reparse the encoded header